File: scripts/backtest/_parse_bt.js
const fs = require('fs');
const buf = fs.readFileSync('D:/QClaw_Trading/scripts/backtest/_bt_out.txt');
const text = buf.toString('latin1');
const lines = text.split('\n').map(l => l.trim()).filter(l => {
  if (!l) return false;
  const pct = (l.match(/%/g) || []).length;
  if (pct >= 10) return false;
  return l.includes('S') || l.includes('回测') || l.includes('夏普') || l.includes('年化') || l.includes('波动率') || l.includes('Top') || l.includes('完整') || l.includes('恒星') || l.includes('纳斯达克') || l.includes('易方达') || l.includes('华夏') || l.includes('广发') || l.includes('华泰') || l.includes('━━') || l.includes('══') || l.includes('基准') || l.includes('胜率') || l.includes('回撤') || l.includes('交易数');
});
console.log(lines.join('\n'));

File: scripts/backtest/_parse_bt.py
# -*- coding: gbk -*-
data = open(r'D:\QClaw_Trading\scripts\backtest\_bt_out.txt','rb').read()
text = data.decode('gbk','ignore')
lines = [l.strip() for l in text.split('\n') if l.strip()]

# Filter out progress bars
filtered = []
skip_patterns = ['Pp%', '═══', '━━━━']
for l in lines:
    # Skip if mostly progress bars
    stripped = l.replace(' ','')
    pct_count = stripped.count('%') + stripped.count('P')
    if pct_count >= 10: continue
    if any(p in l for p in skip_patterns): filtered.append(l); continue
    if l.startswith('P'): continue
    # Keep meaningful lines
    filtered.append(l)

print('\n'.join(filtered))
